src/main/java/it/gov/pagopa/pu/debtpositions/service/create/debtposition/DebtPositionProcessorServiceImpl.java

```diff
@@ -18,8 +20,25 @@ public void updateAmounts(DebtPositionDTO debtPositionDTO) {
     });
   }
 
+  @Override
+  public void updateAmounts(DebtPosition debtPosition) {
+    debtPosition.getPaymentOptions().forEach(paymentOption -> {
+      long totalPaymentOptionAmount = paymentOption.getInstallments().stream()
+        .filter(this::isInstallmentCancelled)
+        .mapToLong(InstallmentNoPII::getAmountCents)
+        .sum();
+
+      paymentOption.setTotalAmountCents(totalPaymentOptionAmount);
+    });
+  }
```

Comment:
let's not introduce a new method, instead let's change the existent method using the common interfaces in the following way:
```suggestion
  @Override
  public void updateAmounts(BaseDebtPosition debtPosition) {
    debtPosition.getPaymentOptions().forEach(paymentOption -> {
      long totalPaymentOptionAmount = paymentOption.getInstallments().stream()
        .filter(this::isInstallmentCancelled)
        .mapToLong(InstallmentNoPII::getAmountCents)
        .sum();

      paymentOption.setTotalAmountCents(totalPaymentOptionAmount);
    });
  }
```

If you found methods not defined in the common interface, you can add them!